addons/sourcemod/scripting/include/Spectate.inc

```diff
@@ -8,6 +8,21 @@
 #endif
 #define _Spectate_included
 
+/**
+ * Called right before a client was force suicide before moving to spec
+ */
+forward void Spectate_OnCommitSuicide(int client);
```

Comment:
you can hook player_death with HookEvent to know this:
- https://sourcemod.dev/#/events/function.HookEvent
- https://wiki.alliedmods.net/Generic_Source_Events#player_death

Reply:
It's called before, not after, that why I need a forward and not hooking the player death. I need this for this case:
https://github.com/Rushaway/ZLeader-Remake/blob/3.4/addons/sourcemod/scripting/ZLeader.sp#L551C1-L558C2

Reply:
so you want to let the player lead if he goes to spec?
You can hook `player_team` and check after the player commits suicide if he has gone to spectator